Sources/Lumina/Camera/LuminaCamera.swift

```diff
@@ -0,0 +1,286 @@
+//
+
+import UIKit
+import AVFoundation
+import CoreML
+
+protocol LuminaCameraDelegate: class {
+  func stillImageCaptured(camera: LuminaCamera, image: UIImage, livePhotoURL: URL?, depthData: Any?)
+  func videoFrameCaptured(camera: LuminaCamera, frame: UIImage)
+  @available (iOS 11.0, *)
+  func videoFrameCaptured(camera: LuminaCamera, frame: UIImage, predictedObjects: [LuminaRecognitionResult]?)
+  func depthDataCaptured(camera: LuminaCamera, depthData: Any)
+  func videoRecordingCaptured(camera: LuminaCamera, videoURL: URL)
+  func finishedFocus(camera: LuminaCamera)
+  func detected(camera: LuminaCamera, metadata: [Any])
+  func cameraSetupCompleted(camera: LuminaCamera, result: CameraSetupResult)
+  func cameraBeganTakingLivePhoto(camera: LuminaCamera)
+  func cameraFinishedTakingLivePhoto(camera: LuminaCamera)
+}
+
+enum CameraSetupResult: String {
+  typealias RawValue = String
+  case videoPermissionDenied = "Video Permissions Denied"
+  case videoPermissionRestricted = "Video Permissions Restricted"
+  case videoRequiresAuthorization = "Video Permissions Require Authorization"
+  case audioPermissionDenied = "Audio Permissions Denied"
+  case audioPermissionRestricted = "Audio Permissions Restricted"
+  case audioRequiresAuthorization = "Audio Permissions Require Authorization"
+  case unknownError = "Unknown Error"
+  case invalidVideoDataOutput = "Invalid Video Data Output"
+  case invalidVideoFileOutput = "Invalid Video File Output"
+  case invalidVideoMetadataOutput = "Invalid Video Metadata Output"
+  case invalidPhotoOutput = "Invalid Photo Output"
+  case invalidDepthDataOutput = "Invalid Depth Data Output"
+  case invalidVideoInput = "Invalid Video Input"
+  case invalidAudioInput = "Invalid Audio Input"
+  case requiresUpdate = "Requires AV Update"
+  case videoSuccess = "Video Setup Success"
+  case audioSuccess = "Audio Setup Success"
+}
+
+public enum TorchState {
+  //swiftlint:disable identifier_name
+  case on(intensity: Float)
+  case off
+  case auto
+}
+
+final class LuminaCamera: NSObject {
+  weak var delegate: LuminaCameraDelegate?
+
+  var torchState: TorchState = .off {
+    didSet {
+      guard let input = self.videoInput else {
+        torchState = .off
+        return
+      }
+      do {
+        try input.device.lockForConfiguration()
+        switch torchState {
+          case .on(let intensity):
+            if input.device.isTorchModeSupported(.on) {
+              try input.device.setTorchModeOn(level: intensity)
+              LuminaLogger.notice(message: "torch mode set to on with intensity: \(intensity)")
+              input.device.unlockForConfiguration()
+            }
+          case .off:
+            if input.device.isTorchModeSupported(.off) {
+              input.device.torchMode = .off
+              LuminaLogger.notice(message: "torch mode set to off")
+              input.device.unlockForConfiguration()
+            }
+          case .auto:
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/dokun1/Lumina/blob/f3c4aa8d5c66dd9d2660cdba8e658b8902c9752f/Sources/Lumina/Camera/LuminaCamera.swift#L79"></span>Case statements should vertically align with their enclosing switch statement.<br />
<code>switch_case_alignment</code> <code>LuminaCamera.swift:79</code></td>
    </tr>
  </tbody>
</table>